### ISSUE TITLE
When one worker node is Powered OFF and then upgrade is initiated, upgrade exits (max_unavailable is provided as a percentage)

### ISSUE THREAD

**Issue by soumyalj** [rancher/rke#1908]
RKE version: v1.1.0-rc5

Docker version: (docker version,docker info preferred) 19.03

Operating system and kernel: (cat /etc/os-release, uname -r preferred) Ubuntu

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**EC2

Steps to Reproduce:

1. Create a RKE Cluster with 1 control plane, 1 etcd node and 5 worker nodes(NW1 to NW5) using K8s version "v1.16.6-rancher1-1"
2. Power OFF one worker node NW2
3. Upgrade to a new K8s version v1.17.2-rancher1-1 using the below upgrade strategy

```
kubernetes_version: v1.17.2-rancher1-1

upgrade_strategy:
  max_unavailable: 40%
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 120

services:
   kubelet:
     extra_args:
       max-pods: 170

```
Upgrade fails with the below error:
```
DEBU[0018] FIXME: Got an status-code for which error does not match any expected type!!!: -1  module=api status_code=-1
WARN[0018] Failed to set up SSH tunneling for host [abcd]: Can't retrieve Docker Info: error during connect: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info: Unable to access node with address [abcd:22] using SSH. Please check if the node is up and is accepting SSH connections or check network policies and firewall rules. Error: dial tcp abcd:22: connect: operation timed out 
WARN[0018] Removing host [abcd] from node lists 
DEBU[0018] Provided value for maxUnavailable: {1 0 40%} 
INFO[0018] 1 worker nodes can be unavailable at a time  
FATA[0018] cannot proceed with upgrade of worker components since 1 (>=maxUnavailable) hosts are inactive; found inactive hosts: abcd 

```

In the above case with with one worker node OFF, 4 worker nodes are still ON, upgrade should still continue and complete for the other nodes. But upgrade exits.

**Comment by soumyalj** [comment]
The same issue is observed when one node is powered OFF and upgrade is initiated, but only kubelet extra_args is changed and K8s version remains the same .
```
upgrade_strategy:
  max_unavailable: 40%
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 120

services:
   kubelet:
     extra_args:
       max-pods: 180
```

RKE logs:
```
WARN[0000] This is not an officially supported version (v1.1.0-rc5) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc5              
INFO[0000] Initiating Kubernetes cluster                
INFO[0000] [certificates] Generating admin certificates and kubeconfig 
INFO[0000] Successfully Deployed state file at [./clusterzeroworker.rkestate] 
INFO[0000] Building Kubernetes cluster                  
INFO[0000] [dialer] Setup tunnel for host [abcd] 
...
....
...
...
....
INFO[0000] [dialer] Setup tunnel for host [abcd] 
WARN[0019] Failed to set up SSH tunneling for host [abcd]: Can't retrieve Docker Info: error during connect: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info: Unable to access node with address [abcd:22] using SSH. Please check if the node is up and is accepting SSH connections or check network policies and firewall rules. Error: dial tcp abcd:22: connect: operation timed out 
WARN[0019] Removing host [abcd] from node lists  
INFO[0019] 1 worker nodes can be unavailable at a time  
FATA[0019] cannot proceed with upgrade of worker components since 1 (>=maxUnavailable) hosts are inactive; found inactive hosts: abcd 




```

**Comment by mrajashree** [comment]
Available with v1.1.0-rc7

**Comment by soumyalj** [comment]
Tested with rke version v1.1.0-rc7

1. Create a RKE Cluster with 1 control plane, 1 etcd node and 5 worker nodes(NW1 to NW5) using K8s version "v1.16.7-rancher1-1"
2. Power OFF one worker node NW2
Upgrade to a new K8s version v1.17.2-rancher1-1 using the below upgrade strategy
kubernetes_version: v1.17.3-rancher1-1
```
upgrade_strategy:
  max_unavailable: 40%
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 120

services:
   kubelet:
     extra_args:
       max-pods: 170
```

Upgrade continues further without any error. Worker nodes other than the powered OFF node upgrade successfully. Logs provide a message that the powered OFF node was skipped as below:

```
INFO[0204] [network] Setting up network plugin: canal   
INFO[0204] [addons] Saving ConfigMap for addon rke-network-plugin to Kubernetes 
INFO[0204] [addons] Successfully saved ConfigMap for addon rke-network-plugin to Kubernetes 
INFO[0204] [addons] Executing deploy job rke-network-plugin 
INFO[0204] [addons] Setting up coredns                  
INFO[0204] [addons] Saving ConfigMap for addon rke-coredns-addon to Kubernetes 
INFO[0204] [addons] Successfully saved ConfigMap for addon rke-coredns-addon to Kubernetes 
INFO[0204] [addons] Executing deploy job rke-coredns-addon 
INFO[0215] [addons] CoreDNS deployed successfully..     
INFO[0215] [dns] DNS provider coredns deployed successfully 
INFO[0215] [addons] Setting up Metrics Server           
INFO[0215] [addons] Saving ConfigMap for addon rke-metrics-addon to Kubernetes 
INFO[0215] [addons] Successfully saved ConfigMap for addon rke-metrics-addon to Kubernetes 
INFO[0215] [addons] Executing deploy job rke-metrics-addon 
INFO[0221] [addons] Metrics Server deployed successfully 
INFO[0221] [ingress] Setting up nginx ingress controller 
INFO[0221] [addons] Saving ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0221] [addons] Successfully saved ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0221] [addons] Executing deploy job rke-ingress-controller 
INFO[0221] [ingress] ingress controller nginx deployed successfully 
INFO[0221] [addons] Setting up user addons              
INFO[0221] [addons] no user addons defined              
FATA[0221] Provisioning incomplete, host(s) [18.220.253.4] skipped because they could not be contacted 

```
